test/test_simple_commands.py

```diff
@@ -78,404 +87,405 @@
 }
 
 
-def pytest_generate_tests(metafunc):
-    # Source: https://docs.pytest.org/en/latest/example/parametrize.html#a-quick-port-of-testscenarios  # noqa
-    idlist = []
-    argvalues = []
-    for scenario in metafunc.cls.scenarios:
-        idlist.append(scenario[0])
-        items = scenario[1].items()
-        argnames = [x[0] for x in items]
-        argvalues.append([x[1] for x in items])
-    metafunc.parametrize(argnames, argvalues, ids=idlist, scope="class")
-
-
-class TestSimpleCommands(OpenWPMTest):
-    """Test correctness of simple commands and check
-    that resulting data is properly keyed.
-
-    This entire test class is parametrized to run against
-    both headless and xvfb modes to ensure both are exercized
-    during the test suite and there are no obvious problems.
+scenarios = [
+    pytest.param("headless", id="headless"),
+    pytest.param("xvfb", id="xvfb"),
+]
+
+
+@pytest.mark.parametrize("display_mode", scenarios)
```

Comment:
Is there a reason we can't use this same parameterize shortcut in `test_memory_storage_provider.py`?

Reply:
We can. And it's a lot cleaner that way. Thanks for pointing this out.